src/k8s/pkg/k8sd/api/kubeconfig.go

```diff
@@ -1,25 +1,60 @@
 package api
 
 import (
+	"encoding/json"
 	"fmt"
 	"net/http"
-	"os"
 
 	apiv1 "github.com/canonical/k8s/api/v1"
+	"github.com/canonical/k8s/pkg/k8sd/api/impl"
+	"github.com/canonical/k8s/pkg/k8sd/setup"
+	"github.com/canonical/k8s/pkg/utils"
 	"github.com/canonical/lxd/lxd/response"
 	"github.com/canonical/microcluster/state"
+	"k8s.io/client-go/tools/clientcmd"
 )
 
 func getKubeconfig(s *state.State, r *http.Request) response.Response {
-	// TODO: Render a new kubeconfig instead of reading the existing one
-	//       when the config can be altered via request parameters.
-	config, err := os.ReadFile("/etc/kubernetes/admin.conf")
+	req := apiv1.GetKubeConfigRequest{}
+	if err := json.NewDecoder(r.Body).Decode(&req); err != nil {
+		return response.BadRequest(fmt.Errorf("failed to parse request: %w", err))
+	}
+
+	// Fetch pieces needed to render an admin kubeconfig: ca, server, token
+	clusterCfg, err := utils.GetClusterConfig(r.Context(), s)
+	if err != nil {
+		return response.InternalError(fmt.Errorf("failed to retrieve cluster config: %w", err))
+	}
+	ca := clusterCfg.Certificates.CACert
+	server := req.Server
+	if req.Server == "" {
+		server = fmt.Sprintf("%s:%d", s.Address().Hostname(), clusterCfg.APIServer.SecurePort)
+	}
+	token, err := impl.GetOrCreateAuthToken(s.Context, s, "kubernetes-admin", []string{"system:masters"})
```

Comment:
assumes we always want this to return the admin kubeconfig.  any value in adding user/group as request params at the moment?